java/java-impl/src/com/intellij/codeInsight/FunctionalInterfaceSuggester.java

```diff
@@ -276,6 +269,20 @@ else if (expression instanceof PsiMethodReferenceExpression) {
                 }
               }
             }
+            else if (maybeDefaultConstructor(referenceExpression, parameters, element)) {
```

Comment:
This code is meant to show the idea: since [`MethodReferenceResolver#resolve`](https://github.com/neshkeev/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/tree/java/MethodReferenceResolver.java#L65) encapsulates the default constructor into `com.intellij.psi.infos.ClassCandidateInfo` it cannot be handled uniformly here, so this if-else-branch fixes the problem.
I require guidance from the **jetbrains** team to make it beter